src/odata-to-abstract-sql.ts

```diff
@@ -760,7 +762,7 @@ export class OData2AbstractSQL {
 				}
 			},
 		);
-		return _.compact(fields);
+		return fields.filter((f): f is NonNullable<typeof f> => !!f);
```

Comment:
Even though _.compact treats empty strings and null/undefined the same, should we just do v even though it's not strictly equivalent?
```suggestion
		return fields.filter((f): f is NonNullable<typeof f> => f != null);
```

Reply:
`!= null` is fine since we can see right above that it's either an array or undefined, what I'd actually do though is chain this directly on as
```ts
return match.map(...).filter(...)

Reply:
Right :sw